src/index.c

```diff
@@ -3185,6 +3212,7 @@ static int git_index_read_iterator(
 
 	clear_uptodate(index);
 
+	index->dirty = 0;
```

Comment:
Is this correct? I cannot see us saving the index to disk here, so I'd consider it dirty

Reply:
Yes, I agree.  I've changed it.